hedera-node/src/main/java/com/hedera/services/stream/RecordsRunningHashLeaf.java

```diff
@@ -65,6 +81,9 @@ public void serialize(final SerializableDataOutputStream out) throws IOException
 		try {
 			// should wait until runningHash has been calculated and set
 			out.writeSerializable(currentRunningHash(), true);
+			out.writeSerializable(nMinus1RunningHash.getHash(), true);
```

Comment:
Are we _certain_ the `FutureHash`'s of the preceding hashes will always be set if the `runningHash`'s future is set? (It certainly **seems** that way!)